torch/fx/node.py

```diff
@@ -32,29 +32,10 @@ def __init__(self, graph: 'Graph', name: str, op: str, target: Target,
         # being invoked, e.g add, layer1, or torch.add
         self.args = args
         self.kwargs = kwargs
-        self.uses = 0
-
-    def find_uses(self) -> List['Node']:
-        """
-        Find all nodes that use the value produced by `self`. The complexity of
-        this function is linear in the number of nodes * number of arguments to
-        each node.
-
-        Note that len(find_uses()) is not necessarily equal to attribute `uses`.
-        This node could be used multiple times in the same `Node`. In that case,
-        the user node would appear once in the return value here, but `uses` would
-        account for the total number of times this Node is used by the user node.
-        e.g. a node for `x + x` would have two uses for the `x` node, but the
-        `x + x` node would appear once in the return from `find_uses`
-        """
-        use_nodes : List[Node] = []
-        for node in self.graph._nodes:
-            def record_use(arg_node : Node) -> None:
-                if arg_node == self and (len(use_nodes) == 0 or use_nodes[-1] != node):
-                    use_nodes.append(node)
-            map_arg(node.args, record_use)
-            map_arg(node.kwargs, record_use)
-        return use_nodes
+        # All of the nodes that use the value produced by this Node
```

Comment:
assigning to `args` and `kwargs` also requires adjustment of the use lists. If you can't assign to them it makes it very hard to patch up the graph.

Reply:
What do you think is a good solution? I could make them properties and automatically update `users` based on the difference of the old and new values. Is that good?